Preprocess steamid, publisherclass, and aicontent Columns

In [13]:
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.preprocessing import StandardScaler
from sklearn.preprocessing import OneHotEncoder

In [14]:
df = pd.read_csv('/content/gamalytic_steam_games.csv')

In [15]:
df.shape

(93338, 6)

In [16]:
df.head()

steamId  price  copiesSold publisherClass  reviewScore  aiContent
0      730    0.0   302158048            AAA           87        NaN
1      570    0.0   212896574            AAA           82        NaN
2   578080    0.0   161971233            AAA           59        NaN
3      440    0.0    99060457            AAA           90        NaN
4  1172470    0.0    67554185            AAA           67        NaN

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93338 entries, 0 to 93337
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   steamId         93338 non-null  int64  
 1   price           93338 non-null  float64
 2   copiesSold      93338 non-null  int64  
 3   publisherClass  93338 non-null  object 
 4   reviewScore     93338 non-null  int64  
 5   aiContent       0 non-null      float64
dtypes: float64(2), int64(3), object(1)
memory usage: 4.3+ MB


In [18]:
#checking duplicate rows
df.duplicated().sum()

np.int64(0)

#publisher class column

In [19]:
df['publisherClass'].value_counts()

publisherClass
Hobbyist    52161
Indie       37304
AA           2844
AAA          1029
Name: count, dtype: int64

 As dataset rows equals 93338 i think we can replace AA and AAA columns with only one column for others and this also may handle unseen data in testing

#Using one hot encoding

handle_unknown='ignore'=>makes model at the training has the ability to transform unseen data and model can effectively process new data, even if it contains previously unseen categories.

In [20]:
# Applying one-hot encoding
encoder = OneHotEncoder(handle_unknown='ignore')
encoded_data = encoder.fit_transform(df[['publisherClass']]).toarray()

In [21]:
encoded_data = pd.DataFrame(encoded_data, columns=encoder.get_feature_names_out(['publisherClass']))
df = pd.concat([df, encoded_data], axis=1)

In [22]:
df.drop('publisherClass', axis=1, inplace=True)

In [23]:
df.head()

steamId  price  copiesSold  reviewScore  aiContent  publisherClass_AA  \
0      730    0.0   302158048           87        NaN                0.0   
1      570    0.0   212896574           82        NaN                0.0   
2   578080    0.0   161971233           59        NaN                0.0   
3      440    0.0    99060457           90        NaN                0.0   
4  1172470    0.0    67554185           67        NaN                0.0   

   publisherClass_AAA  publisherClass_Hobbyist  publisherClass_Indie  
0                 1.0                      0.0                   0.0  
1                 1.0                      0.0                   0.0  
2                 1.0                      0.0                   0.0  
3                 1.0                      0.0                   0.0  
4                 1.0                      0.0                   0.0

Dropping aicontent column as it contain 100% NULL values

In [24]:
df.drop('aiContent', axis=1, inplace=True)